services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -3114,4 +2864,112 @@ private OFMeterConfig getMeter(DatapathId dpid, long meter) throws SwitchOperati
+        }
+        pushFlow(sw, flowName, switchFlowTuple.getFlow());
+    }
+
+    private void processMeter(IOFSwitch sw, OFMeterMod meterMod) {
+        long meterId = meterMod.getMeterId();
+        OFMeterConfig meterConfig;
+        try {
+            meterConfig = getMeter(sw.getId(), meterId);
+        } catch (SwitchOperationException e) {
+            logger.warn("Meter {} won't be installed on the switch {}: {}", meterId, sw.getId(), e.getMessage());
+            return;
+        }
+
+        OFMeterBandDrop meterBandDrop = Optional.ofNullable(meterConfig)
+                .map(OFMeterConfig::getEntries)
+                .flatMap(entries -> entries.stream().findFirst())
+                .map(OFMeterBandDrop.class::cast)
+                .orElse(null);
+
+        try {
+            OFMeterBandDrop ofMeterBandDrop = sw.getOFFactory().getVersion().compareTo(OF_13) > 0
+                    ? (OFMeterBandDrop) meterMod.getBands().get(0) : (OFMeterBandDrop) meterMod.getMeters().get(0);
+            long rate = ofMeterBandDrop.getRate();
+            Set<OFMeterFlags> flags = meterMod.getFlags();
+
+            if (meterBandDrop != null && meterBandDrop.getRate() == rate
+                    && CollectionUtils.isEqualCollection(meterConfig.getFlags(), flags)) {
+                logger.debug("Meter {} won't be reinstalled on switch {}. It already exists", meterId, sw.getId());
+                return;
+            }
+
+            if (meterBandDrop != null) {
+                logger.info("Meter {} with origin rate {} will be reinstalled on {} switch.",
+                        meterId, sw.getId(), meterBandDrop.getRate());
+                buildAndDeleteMeter(sw, sw.getId(), meterId);
+                sendBarrierRequest(sw);
+            }
+
+            installMeterMod(sw, meterMod);
+        } catch (SwitchOperationException e) {
+            logger.warn("Failed to (re)install meter {} on switch {}: {}", meterId, sw.getId(), e.getMessage());
+        }
+    }
+
+    private void installMeterMod(IOFSwitch sw, OFMeterMod meterMod) throws OfInstallException {
+        logger.info("Installing meter {} on switch {}", meterMod.getMeterId(), sw.getId());
+
+        pushFlow(sw, "--InstallMeter--", meterMod);
+
+        // All cases when we're installing meters require that we wait until the command is processed and
+        // the meter is installed.
+        sendBarrierRequest(sw);
+    }
+
+    private void processGroup(IOFSwitch sw, OFGroupAdd groupAdd) {
+        try {
+            installRoundTripLatencyGroup(sw, groupAdd);
+            logger.debug("Round trip latency group was installed on switch {}", sw.getId());
+        } catch (OfInstallException | UnsupportedOperationException e) {
+            String message = String.format(
+                    "Couldn't install round trip latency group on switch %s. "
+                            + "Standard discovery actions will be installed instead. Error: %s", sw.getId(),
```

Comment:
log messages has details about "round trip latency group". But in common case some other groups can be installed here

Reply:
Unfortunately not only logs here are related to "round trip latency group". Validation here are also correct only for this group use case. For now we have only one group (this one) so I haven't worked on generalized group processing in this patch.